includes/class-wc-payment-gateway-wcpay.php

```diff
@@ -1097,6 +1097,17 @@ public function process_payment_for_order( $cart, $payment_information, $additio
 				$additional_api_parameters['is_platform_payment_method'] = 'true';
 			}
 
+			$statement_descriptor                  = $this->get_account_statement_descriptor();
+			$short_statement_descriptor            = ! empty( $this->get_option( 'short_statement_descriptor' ) ) ? str_replace( "'", '', $this->get_option( 'short_statement_descriptor' ) ) : '';
+			$is_short_statement_descriptor_enabled = ! empty( $this->get_option( 'is_short_statement_descriptor_enabled' ) ) && 'yes' === $this->get_option( 'is_short_statement_descriptor_enabled' );
+			$descriptor                            = null;
+			if ( in_array( $order->get_payment_method(), [ 'card', 'woocommerce_payments' ], true ) && $is_short_statement_descriptor_enabled && ! empty( $short_statement_descriptor ) ) {
+				// Use the shortened statement descriptor for card transactions only.
+				$descriptor = WC_Payments_Utils::get_dynamic_statement_descriptor( $short_statement_descriptor, $order );
+			} elseif ( ! empty( $statement_descriptor ) ) {
+				$descriptor = WC_Payments_Utils::clean_statement_descriptor( $statement_descriptor );
+			}
+
```

Comment:
Thanks for suggesting that, it makes more sense. I've applied it in c342dcb. I only changed the return in the `elseif` to return `$full_statement_descriptor` as per our discussion below.

I went through all the testing with UPE enabled and seems to work fine, I'd appreciate if you can make another round of testing, specially with UPE, since we've changed the first condition (compared to the previous in the UPE gateway) to make the approach reusable.